### ISSUE TITLE
Installation failures: dpkg is locked by another process

### ISSUE THREAD

**Issue by Silent-work** [lancefrench/oms-agent-for-linux-boshrelease#7]
We've been encountering multiple VMs failing their deployments because `omsagent` and `omi` are not able to start. After reviewing the logs, we found that `dpkg` was locked when it was suppose to install `omsagent`, `omi`, `scx` and `omsconfig`. Here was all the output in `pre-start.stderr.log`

```
/var/vcap/jobs/omsagent/bin/pre-start: line 74: scxadmin: command not found
Failed to stop omid.service: Unit omid.service not loaded.
dpkg: error: dpkg frontend is locked by another process
usermod: user 'nxautomation' does not exist
dpkg: error: dpkg frontend is locked by another process
cp: cannot stat '/etc/rsyslog.d/95-omsagent.conf': No such file or directory
Failed to stop omsagent-<redacted>.service: Unit omsagent-<redacted>.service not loaded.
Failed to stop omsagent-<redacted>.service: Unit omsagent-<redacted>.service not loaded.
Failed to execute operation: No such file or directory
Failed to stop omid.service: Unit omid.service not loaded.
Failed to stop omid.service: Unit omid.service not loaded.
Failed to execute operation: No such file or directory
```

Because `dpkg` was locked, the packages weren't actually installed, and because we have a `|| true` after the `dpkg install`, if the installation fails, it returns true instead which allows `pre-start` script to continue. This in turns leads to the VM being in a failing state with `monit` constantly trying to start `omi` and `omsagent`, but those fail because the packages aren't actually there.

The current code block I'm referring to is from the file `jobs/omsagent/templates/install.erb`:

```
# Install packages
if [ ${DPKG_FOUND} -eq 0 ]; then
    (
    flock -x 200
    dpkg -i --force-confold --force-confdef /var/vcap/packages/omsagent/omi*.deb /var/vcap/packages/omsagent/scx*.deb /var/vcap/packages/omsagent/omsagent*.deb /var/vcap/packages/omsagent/omsconfig*.deb || true
    ) 200>/var/vcap/data/dpkg.lock
else
    rpm -Uvh --force /var/vcap/packages/omsagent/omi-*.rpm /var/vcap/packages/omsagent/scx*.rpm /var/vcap/packages/omsagent/omsagent*.rpm /var/vcap/packages/omsagent/omsconfig*.rpm || true
fi
```

**Comment by lancefrench** [comment]
Suggested workaround from the interwebs to see if installs are in progress:

```
#!/bin/sh

while fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
   sleep 1
done
```

We'd probably want to add timeout logic per our offline discussion.

**Comment by lancefrench** [comment]
Merged code changes in #8
